ghost/collections/src/CollectionsService.ts

```diff
@@ -1,41 +1,65 @@
 import {Collection} from './Collection';
+import {PostDTO} from './PostDTO';
 
 type CollectionsServiceDeps = {
-    repository: any;
+    collectionsRepository: any;
+    postsRepository: {
+        getBulk: any;
+    }
 };
 
 export class CollectionsService {
-    repository: any;
+    collectionsRepository: any;
+    postsRepository: any;
 
     constructor(deps: CollectionsServiceDeps) {
-        this.repository = deps.repository;
+        this.collectionsRepository = deps.collectionsRepository;
+        this.postsRepository = deps.postsRepository;
     }
 
     async save(data: any): Promise<Collection> {
         const collection = await Collection.create(data);
-        await this.repository.save(collection);
+        await this.collectionsRepository.save(collection);
+        return collection;
+    }
+
+    /**
+     *
+     * @param collection to add tags to
+     * @param postIds
+     */
+    private async addPosts(collection: Collection, postDTOs: PostDTO[]) : Promise<Collection> {
+        const postIds = postDTOs.map(post => post.id);
+        const posts = await this.postsRepository.getBulk(postIds);
+
+        collection.posts = posts.map((post: any) => post.id);
```

Comment:
I think `collection.posts` should be of type `PostDTO[]` but this sets it to `string[]`